src/System.Device.Gpio/System/Device/Gpio/ThrowHelper.cs

```diff
@@ -0,0 +1,91 @@
+        }
+
+        public static void ThrowInvalidOperationException(ExceptionResource resource, int errorCode = -1, int pin = -1)
+        {
+            throw GetInvalidOperationException(GetResourceString(resource, errorCode, pin));
+        }
+
+        public static PlatformNotSupportedException GetPlatformNotSupportedException(ExceptionResource resource)
+        {
+            return new PlatformNotSupportedException(GetResourceString(resource, -1, -1));
+        }
+
+        private static IOException GetIOException(string message)
+        {
+            return new IOException(message);
+        }
+
+        private static InvalidOperationException GetInvalidOperationException(string message)
+        {
+            return new InvalidOperationException(message);
+        }
+
+        public static InvalidOperationException GetInvalidOperationException(ExceptionResource resource, int pin = -1)
+        {
+            return GetInvalidOperationException(GetResourceString(resource, -1, pin));
+        }
+
+        private static string GetResourceString(ExceptionResource resource, int errorCode, int pin)
+        {
+            string message = "";
+            switch (resource)
+            {
+                case ExceptionResource.NoChipIteratorFound:
+                    message = $"Unable to find a chip iterator, error code: { errorCode}";
+                    break;
+                case ExceptionResource.NoChipFound:
+                    message = $"Unable to find a chip, error code: {errorCode}";
+                    break;
+                case ExceptionResource.PinModeReadError:
+                    message = $"Error while reading pin mode for pin: {pin}";
+                    break;
+                case ExceptionResource.OpenPinError:
+                    message = $"Pin {pin} not available, error: {errorCode}";
+                    break;
+                case ExceptionResource.ReadPinError:
+                    message = $"Error while reading value from pin: {pin}, error: {errorCode}";
+                    break;
+                case ExceptionResource.PinNotOpenedError:
+                    message = $"Can not access pin {pin} that is not open.";
+                    break;
+                case ExceptionResource.SetPinModeError:
+                    message = $"Error setting pin mode, for pin: {pin}, error: {errorCode}";
+                    break;
+                case ExceptionResource.ConvertPinNumberingSchemaError:
+                    message = $"This driver is generic so it cannot perform conversions between pin numbering schemes.";
+                    break;
+                default:
+                    Debug.Fail($"The ExceptionResource enum value: {resource} is not part of the switch. Add the appropriate case and exception message.");
+                    break;
+            }
+            
+            return message;
+        }
+
+        internal enum ExceptionResource
```

Comment:
might be better to make this a static class with const strings, then you don't need the large switch

Reply:
How would that work? The code path with the switch only occurs on exceptions and it is somewhat easy to extend to add new exceptions. I don't think const strings would work since a lot of the messages are format strings with custom args.

Reply:
i think its better to be enum for swith case ...